build-support/bin/common.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+"""Utils for scripts to interface with the outside world."""
+
+import time
+from contextlib import contextmanager
+from typing import Tuple
+
+# --------------------------------------------------------
+# Logging utils
+# --------------------------------------------------------
+
+_SCRIPT_START_TIME = time.time()
+
+_CLEAR_LINE = "\x1b[K"
+_COLOR_BLUE = "\x1b[34m"
+_COLOR_RED = "\x1b[31m"
+_COLOR_GREEN = "\x1b[32m"
+_COLOR_RESET = "\x1b[0m"
+
+
+def die(message: str) -> None:
+  raise SystemExit(f"{_COLOR_RED}{message}{_COLOR_RESET}")
+
+
+def green(message: str) -> None:
+  print(f"{_COLOR_GREEN}{message}{_COLOR_RESET}")
```

Comment:
I'm fine with leaving these helper methods in this file for now in order to unblock others from jumping in and writing python 3 scripts themselves without having to reinvent the wheel, especially since some of this is used in the setup repo.

Reply:
Where else would these go? It's meant to emulate `common.sh`.

One limitation, though, is that this file has to live in `build-support/bin` because Python scripts cannot import files from their parent directory, unless we are willing to accept `sys` path hackery at the top of our `bin/*.py` scripts.

Reply:
I'm fine with it living here, but can you kill all the unused code? you only need `die`, `_COLOR_RED` and `_COLOR_RESET` fwict. This is generally not awesome though as a script. We should probably be dogfooding and just using a custom task and re-using ansicolors, etc. Since isort et all currently don't produce data products though that you could consume and format to output, thats not an easy task at the moment.

Reply:
... and towards this end - all this work at making isort useable is lost on Pants users. It wouldn;t be lost though is fixed via dogfooding.